frame/evm-precompile/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+// SPDX-License-Identifier: Apache-2.0
+// This file is part of Frontier.
+//
+// Copyright (c) 2020-2023 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+pub mod weights;
+pub use weights::*;
+
+use frame_support::BoundedVec;
+use pallet_evm::{Precompile, PrecompileHandle, PrecompileResult, PrecompileSet};
+use pallet_evm_precompile_modexp::Modexp;
+use pallet_evm_precompile_sha3fips::Sha3FIPS256;
+use pallet_evm_precompile_simple::{ECRecover, ECRecoverPublicKey, Identity, Ripemd160, Sha256};
+use scale_codec::{Decode, Encode};
+use scale_info::{prelude::marker::PhantomData, TypeInfo};
+use sp_core::{ConstU32, MaxEncodedLen, H160};
+use sp_std::ops::Deref;
+
+#[derive(Decode, Encode, Default, TypeInfo, Clone, PartialEq, Debug, MaxEncodedLen)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub struct PrecompileLabel(BoundedVec<u8, ConstU32<32>>);
+
+impl PrecompileLabel {
+	pub fn new(l: BoundedVec<u8, ConstU32<32>>) -> PrecompileLabel {
+		PrecompileLabel(l)
+	}
+}
+
+impl Deref for PrecompileLabel {
+	type Target = BoundedVec<u8, ConstU32<32>>;
+
+	fn deref(&self) -> &Self::Target {
+		&self.0
+	}
+}
+
+pub struct OnChainPrecompiles<R>(PhantomData<R>);
+
+impl<R> OnChainPrecompiles<R>
+where
+	R: pallet::Config + pallet_evm::Config,
+{
+	pub fn new() -> Self {
+		Self(Default::default())
+	}
+}
+impl<R> PrecompileSet for OnChainPrecompiles<R>
```

Comment:
`pallet-evm` has the [following types in its `Config` trait](https://github.com/paritytech/frontier/blob/58c568964dc32dd96153b7d113ac2df58d27de2b/frame/evm/src/lib.rs#L138-L139):
```
type PrecompilesType: PrecompileSet;
type PrecompilesValue: Get<Self::PrecompilesType>;
```

as pointed out by @kianenigma , it seems that is not doable, because pallet structs are not meant to be instantiated: https://substrate.stackexchange.com/questions/7579/how-to-implement-the-get-trait-for-a-pallett?noredirect=1#comment7602_7579